File: CMPS3500_MiniScheme_Starter_Package/procedural/file_loader_stub.cpp
// Tiny file-loading helper stub for the procedural implementation.
// Replace or adapt as needed.
#include <fstream>
#include <sstream>
#include <string>

std::string read_file(const std::string& path) {
    std::ifstream in(path);
    //error handling: throw an exception if the file cannot be opened
    if (!in) {
        throw std::runtime_error("Could not open file: " + path);
    }
    std::ostringstream ss;
    ss << in.rdbuf();
    return ss.str();
}

File: CMPS3500_MiniScheme_Starter_Package/oop/FileLoaderStub.java
// Tiny file-loading helper stub for the OOP implementation.
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public class FileLoaderStub {
    public static String readFile(String path) throws Exception {
        try {
        return new String(Files.readAllBytes(Paths.get(path)));
        } catch (Exception e){
            throw new RuntimeException("Could not open file: " + path);
        }
    }
}
